tests/unit/test-cmd/test.build.js

```diff
@@ -329,4 +367,42 @@ describe('build', () => {
 
   });
 
+  describe('defaultPackageCreator', () => {
+    it('should reject on Unexpected errors', () => {
+      return withTempDir(
+        (tmpDir) => {
+          const fakeEventToPromise = sinon.spy(async (stream) => {
+            await defaultEventToPromise(stream, 'close');
+            // Remove contents of tmpDir before removal of directory.
+            const files = await fs.readdir(tmpDir.path());
+            for (const file of files) {
+              await fs.unlink(path.join(tmpDir.path(), file));
+            }
+            return Promise.reject(new Error('Unexpected error'));
+          });
+          const sourceDir = fixturePath('minimal-web-ext');
+          const artifactsDir = tmpDir.path();
+          const fileFilter = new FileFilter({sourceDir, artifactsDir});
+          const params = {
+            manifestData: basicManifest,
+            sourceDir,
+            fileFilter,
+            artifactsDir,
+            overwriteDest: false,
+            showReadyMessage: false,
+          };
+          const options = {
+            eventToPromise: fakeEventToPromise,
+          };
+
+          return defaultPackageCreator(params, options)
+            .then(() => assert.fail('The expected error has not been raised'))
```

Comment:
You can import [this helper](https://github.com/mozilla/web-ext/blob/master/tests/unit/helpers.js#L116) and use it like `.then(makeSureItFails()).catch(...)`